roles/vendors/kvm/defaults/main.yml

```diff
@@ -2,8 +2,6 @@
 boot_iso_url: "{{ discovery_iso_server }}/{{ discovery_iso_name }}"
 secure_sushy_tools: "{{ secure | default(true) }}"
 bmc_address: "{{ hostvars[target_host]['bmc_address'] }}"
-bmc_user: "{{ hostvars[target_host]['bmc_user'] }}"
```

Comment:
The values are already omitted in the tasks, in order for omit to work there you can't pass any value to the role include not even empty string

Reply:
I think there will be an issue with `boot_disk` usage because it isn't delegated from the `target_host`. Losing this redirection will cause the `bmc_user` to be looked up on the `bastion` instead.

So I think it may be better to put the omit here and remove it from the tasks.

Reply:
Not sure omit will work in defaults but let me give it a try

Reply:
I think its implemented as a value so I think it should work.

Reply:
yes but what happens when `target_host` is not defined?

Reply:
Seems to work fine:
```yaml
---
- hosts: localhost
  tasks:
    - debug:
        msg: "This should print {{ hostvars[undef]['ansible_architecture'] | default('not found') }}"
...
```
outputs
```
PLAY [localhost] ****************************************************************************

TASK [Gathering Facts] **********************************************************************
ok: [localhost]

TASK [debug] ********************************************************************************
ok: [localhost] => {
    "msg": "This should print not found"
}

PLAY RECAP **********************************************************************************
localhost                  : ok=2    changed=0    unreachable=0    failed=0    skipped=0    rescued=0    ignored=0   
```

Reply:
I don't think people should use vendors directly they should instead use `boot_iso` or `boot_disk` which set `target_host`.

If we wanted to guard against it we could set a default `target_host: "{{ inventory_hostname }}"` or something like that.

Reply:
> Seems to work fine:
Thanks for checking, I think there are scenarios where it does fail perhaps another layer deep or something like that.